designs/move-cluster-specific-code-out-of-manager.md

```diff
@@ -0,0 +1,209 @@
+Move cluster-specific code out of the manager
+===================
```

Comment:
Can we add these sections to clearly define the scope and the problem statement:

- Motivation
- Goal / non-goals for this proposal
- Requirements